spec/tunes/app_version_spec.rb

```diff
@@ -150,6 +171,303 @@
+      end
+
+      it "deletes the geojson" do
+        version.upload_geojson!(du_fixture_file_path('upload_valid.geojson'))
+        version.upload_geojson!(nil)
+        expect(version.raw_data['transitAppFile']['value']).to eq(nil)
+        expect(version.transit_app_file).to eq(nil)
+      end
+    end
+
+    describe "Upload screenshots", :screenshots do
+      before do
+        allow(Spaceship::UploadFile).to receive(:from_path) do |path|
+          du_uploadimage_correct_screenshot if path == 'path_to_screenshot'
+        end
+      end
+      let(:screenshot_path) { 'path_to_screenshot' }
+
+      describe "Parameter checks" do
+        it "prevents from using negative sort_order" do
+          expect do
+            version.upload_screenshot!(screenshot_path, -1, "English", :iphone4)
+          end.to raise_error "sort_order must be positive"
+        end
+
+        it "prevents from using sort_order 0" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 0, "English", :iphone4)
+          end.to raise_error "sort_order must be positive"
+        end
+
+        it "prevents from using too large sort_order" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 6, "English", :iphone4)
+          end.to raise_error "sort_order must not be > 5"
+        end
+
+        # not really sure if we want to enforce that
+        # it "prevents from letting holes in sort_orders" do
+        #  expect do
+        #    version.upload_screenshot!(screenshot_path, 4, "English", :iphone4)
+        #  end.to raise_error "FIXME"
+        # end
+
+        it "prevent from using invalid language" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 1, "NotALanguage", :iphone4)
+          end.to raise_error "NotALanguage isn't an activated language"
+        end
+
+        it "prevent from using invalid language" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 1, "English_CA", :iphone4)
+          end.to raise_error "English_CA isn't an activated language"
+        end
+
+        it "prevent from using invalid device" do
+          expect do
+            version.upload_screenshot!(screenshot_path, 1, "English", :android)
+          end.to raise_error "android isn't a valid device name"
+        end
+      end
+
+      describe "Add, Replace, Remove screenshots" do
+        it "can add a new screenshot to the list" do
+          du_upload_screenshot_success
+
+          count = version.screenshots['English'].count
+          version.upload_screenshot!(screenshot_path, 3, "English", :iphone4)
+          expect(version.screenshots['English'].count).to eq(count + 1)
+        end
+
+        it "can replace an existing screenshot with existing sort_order" do
+          du_upload_screenshot_success
+
+          count = version.screenshots['English'].count
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.